Terraform uses default kubectl context group configurations rather what is defined in context

Hi there,

Came across an interesting problem. It looks like Terraform by default tries to use the context configurations within the system for kubernetes context authentication. So although you define a context and your kubectl defaults to something else kube related data lookups will fail for a terraform plan. Is there a way to tell Terraform to not use kubectl context in system? Or is this a feature request?

### Terraform Version
Terraform v0.11.7

### Affected Resource(s)
- kubernetes
- kubernetes_secret


### Terraform Configuration Files
```hcl
provider "kubernetes" "project_kube_dga_k8_context" {
  alias                  = "project_kube_dga_k8_context"
  config_context_cluster = "gke_env-kube-east1-b_prod-gke-dga"
}
resource "google_service_account_key" "env_datastore_user_key" {
  service_account_id = "${google_service_account.env_datastore_datastore_user.name}"
}
resource "kubernetes_secret" "kubernetes_dga_datastore_credentials" {
  provider = "kubernetes.project_kube_dga_k8_context"
  metadata {
    name = "env-datastore-user-key"
  }
  data {
    credentials.json = "${base64decode(google_service_account_key.env_datastore_user_key.private_key)}"
  }
}
```

### Debug Output
```
* kubernetes_secret.kubernetes_dga_datastore_credentials: kubernetes_secret.kubernetes_dga_datastore_credentials: secrets "env-datastore-user-key" is forbidden: User "system:anonymous" cannot get secrets in the namespace "default": Unknown user "system:anonymous"
```

### Panic Output
N/A

### Expected Behavior
gke_env-kube-east1-b_prod-gke-dga cluster context should have been used.

### Actual Behavior
Default kubectl cluster context gets used

### Workaround
Simply change the kubectl installed on the system to use the same context

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

1. Set a different cluster context manually for kubectl than what is used by your kubernetes provider config.
2. `terraform apply`

### Important Factoids
- Ran this against GKE cluster, however, I assume this would be same for all kubernetes clusters.

### References
N/A

Same issue here.

I believe the provider should error instead of falling back on system's default configuration if any of the explicitly set configuration parameters is incorrect.

Provider configuration:

```
provider "kubernetes" {
  version = "1.3.0"

  alias = "cluster_regional_1"

  load_config_file = true

  config_path            = "./myproject-preprod-europe-west1-gke1.kubeconfig"
  config_context         = "gke_myproject-preprod_europe-west1_myproject-preprod-europe-west1-gke1"
  config_context_cluster = "gke_myproject-preprod_europe-west1_myproject-preprod-europe-west1-gke1"

  host = "35.190.XXX.XXX"
}

```

Excerpt from plan with missing kubernetes configuration file:

```
2018-10-30T20:26:54.220+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.2.0-3f176651: 2018/10/30 20:26:54 [DEBUG] Using custom current context: "gke_myproject-preprod_europe-west1_myproject-preprod-europe-west1-gke1"
2018-10-30T20:26:54.220+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.2.0-3f176651: 2018/10/30 20:26:54 [DEBUG] Using overidden context: api.Context{LocationOfOrigin:"", Cluster:"gke_myproject-preprod_europe-west1_myproject-preprod-europe-west1-gke1", AuthInfo:"", Namespace:"", Extensions:map[string]runtime.Object(nil)}
2018-10-30T20:26:54.220+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.2.0-3f176651: 2018/10/30 20:26:54 [INFO] Unable to load config file as it doesn't exist at "./myproject-preprod-europe-west1-gke1.kubeconfig"
module.cluster_regional_1.module.namespaces_system.kubernetes_namespace.namespaces[0]: Creating...
  metadata.#:                  "" => "1"
2018-10-30T20:26:54.224+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.2.0-3f176651: 2018/10/30 20:26:54 [INFO] Creating new namespace: v1.Namespace{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"default", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"name":"default"}, Annotations:map[string]string{}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.NamespaceSpec{Finalizers:[]v1.FinalizerName(nil)}, Status:v1.NamespaceStatus{Phase:""}}
  metadata.0.generation:       "" => "<computed>"
  metadata.0.labels.%:         "" => "1"
  metadata.0.labels.name:      "" => "default"
  metadata.0.name:             "" => "default"
  metadata.0.resource_version: "" => "<computed>"
  metadata.0.self_link:        "" => "<computed>"
  metadata.0.uid:              "" => "<computed>"
module.cluster_regional_1.namespaces_system.kubernetes_namespace.namespaces.0: Still creating... (10s elapsed)
module.cluster_regional_1.namespaces_system.kubernetes_namespace.namespaces.0: Still creating... (20s elapsed)
module.cluster_regional_1.namespaces_system.kubernetes_namespace.namespaces.0: Still creating... (30s elapsed)
2018/10/30 20:27:24 [ERROR] root.cluster_regional_1.namespaces_system: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* kubernetes_namespace.namespaces.0: Post http://35.190.XXX.XXX/api/v1/namespaces: dial tcp 35.190.XXX.XXX:80: i/o timeout
2018/10/30 20:27:24 [ERROR] root.cluster_regional_1.namespaces_system: eval: *terraform.EvalSequence, err: 1 error(s) occurred:
```

Excerpt from apply with missing kubernetes configuration file:

```
2018-10-30T20:39:47.314+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.3.0_x4: 2018/10/30 20:39:47 [DEBUG] Using custom current context: "gke_myproject-preprod_europe-west1_myproject-preprod-europe-west1-gke1"
2018-10-30T20:39:47.314+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.3.0_x4: 2018/10/30 20:39:47 [DEBUG] Using overidden context: api.Context{LocationOfOrigin:"", Cluster:"gke_myproject-preprod_europe-west1_myproject-preprod-europe-west1-gke1", AuthInfo:"", Namespace:"", Extensions:map[string]runtime.Object(nil)}
2018-10-30T20:39:47.314+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.3.0_x4: 2018/10/30 20:39:47 [INFO] Unable to load config file as it doesn't exist at "./myproject-preprod-europe-west1-gke1.kubeconfig"
module.cluster_regional_1.module.namespaces_system.kubernetes_namespace.namespaces[0]: Creating...
  metadata.#:                  "" => "1"
  metadata.0.generation:       "" => "<computed>"
  metadata.0.labels.%:         "" => "1"
  metadata.0.labels.name:      "" => "default"
  metadata.0.name:             "" => "default"
  metadata.0.resource_version: "" => "<computed>"
  metadata.0.self_link:        "" => "<computed>"
  metadata.0.uid:              "" => "<computed>"
2018-10-30T20:39:47.320+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.3.0_x4: 2018/10/30 20:39:47 [INFO] Creating new namespace: v1.Namespace{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"default", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"name":"default"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.NamespaceSpec{Finalizers:[]v1.FinalizerName(nil)}, Status:v1.NamespaceStatus{Phase:""}}
module.cluster_regional_1.namespaces_system.kubernetes_namespace.namespaces.0: Still creating... (10s elapsed)
module.cluster_regional_1.namespaces_system.kubernetes_namespace.namespaces.0: Still creating... (20s elapsed)
module.cluster_regional_1.namespaces_system.kubernetes_namespace.namespaces.0: Still creating... (30s elapsed)

2018/10/30 20:40:19 [DEBUG] plugin: waiting for all plugin processes to complete...
Error: Error applying plan:

1 error(s) occurred:

2018-10-30T20:40:19.791+0100 [DEBUG] plugin.terraform-provider-google_v1.19.1_x4: 2018/10/30 20:40:19 [ERR] plugin: plugin server: accept unix /tmp/plugin027115749: use of closed network connection
2018-10-30T20:40:19.792+0100 [DEBUG] plugin.terraform-provider-kubernetes_v1.3.0_x4: 2018/10/30 20:40:19 [ERR] plugin: plugin server: accept unix /tmp/plugin236270776: use of closed network connection
* module.cluster_regional_1.module.namespaces_system.kubernetes_namespace.namespaces[0]: 1 error(s) occurred:

2018-10-30T20:40:19.795+0100 [DEBUG] plugin: plugin process exited: path=/home/pdecat/workspaces/clients/myproject/myproject.terraform.google/myproject-preprod/europe-west1/k8s/.terraform/plugins/linux_amd64/terraform-provider-google_v1.19.1_x4
* kubernetes_namespace.namespaces.0: Post http://35.190.XXX.XXX/api/v1/namespaces: dial tcp 35.190.XXX.XXX:80: i/o timeout
2018-10-30T20:40:19.795+0100 [DEBUG] plugin: plugin process exited: path=/home/pdecat/workspaces/clients/myproject/myproject.terraform.google/myproject-preprod/europe-west1/k8s/.terraform/plugins/linux_amd64/terraform-provider-kubernetes_v1.3.0_x4

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

Note that it only fails because I explicitely set master IP address in `host` field.
Otherwise, it would have created the resource in the system's default configured cluster.
I voluntarily set the most fields to catch this issue.

--- a/website/docs/index.html.markdown
+++ b/website/docs/index.html.markdown
@@ -16,8 +16,7 @@ Use the navigation to the left to read about the available resources.
 
 ```hcl
 provider "kubernetes" {
-  config_context_auth_info = "ops"
-  config_context_cluster   = "mycluster"
+  config_context = "my-context"
 }
 
 resource "kubernetes_namespace" "example" {

